### PR TITLE
fix: exit success for graceful detaches

### LINKED ISSUE #130: non-zero exit status upon `shpool detach`
### What happened

```sh
$ shpool attach main
shpool:main $ shpool detach
$ echo $?
1
```

### What I expected to happen

I expect `shpool` to exit 0 when explicitly detaching from a session. This makes the following auto-attach setup smoother:

```
# ~/.ssh/authorized_keys
command="XDG_RUNTIME_DIR=/run/user/1000 $HOME/.cargo/bin/shpool attach main || bash" ssh-rsa AAAAB3N...
```

(In this setup, I would like `shpool detach` to drop me out of the SSH session entirely, but it falls through to a new bash prompt instead.)

### Version info

0.6.3

### Logs

N/A? (happy to add them if you aren't getting the same result for some reason.)

### PR DESCRIPTION
**Pull request by ethanpailes**
This patch fixes the exit code for the attach
process so that it is 0 when shpool has been
told to gracefully detach, either via the
`shpool detach` subcommand or via the detach
keybinding. It is more semantically correct
to exit 0 in these cases.

Fixes #130

BREAKING CHANGE: technically could impact some scripts

### CODE REVIEW

File: shpool/tests/attach.rs [hunk 1]
```diff
@@ -554,7 +554,8 @@ fn default_keybinding_detach() -> anyhow::Result<()> {
         lm1.scan_until_re("someval$")?;
 
         a1.run_raw_cmd(vec![0, 17])?; // Ctrl-Space Ctrl-q
-        a1.proc.wait()?;
+        let exit_status = a1.proc.wait()?;
+        assert!(exit_status.success());
```
Comment (jrmlhermitte) [inline shpool/tests/attach.rs:558]: just curious, does this run at runtime for users? (I assume this is like python assert which only runs if debug mode is set i think)

File: shpool/tests/detach.rs [hunk 2]
```diff
@@ -34,6 +34,9 @@ fn single_running() -> anyhow::Result<()> {
 
         daemon_proc.events = Some(waiter.wait_final_event("daemon-bidi-stream-done")?);
 
+        let attach_exit_status = attach_proc.proc.wait()?;
+        assert!(attach_exit_status.success());
```
Comment (jrmlhermitte) [inline shpool/tests/detach.rs:38]: dumb question (to save me time), this looks to return an unwraped i32. where does the `.success()` method come from on this type?
If we're using traits somehow, should we not define an i32 type but rather make a new type?
I may have gotten all this wrong. 

for this comment, it is ok if `.success()` means we have exit status 0 and not other way around.
(ditto for another similar line, given this is only two lines i wont add a ccmment to it)

**Review by jrmlhermitte — APPROVED** [review]

**Comment by jrmlhermitte** [comment]
On more comment, should we add a line to test we're getting zero exit status somewhere, do we have an existing test we can use? I'm thinking less of testing this now, but ensuring that future changes don't undo this since this is a breaking change

File: shpool/tests/detach.rs (hunk 2 above)
Comment (ethanpailes) [reply shpool/tests/detach.rs:38]: The stdlib has an `ExitStatus` type that has a few methods hanging off of it. You can unwrap to the inner int if you want, but for just checking success you can use the methods.

File: shpool/tests/attach.rs (hunk 1 above)
Comment (ethanpailes) [reply shpool/tests/attach.rs:558]: No it is never compiled out (there is [debug_assert!](https://doc.rust-lang.org/std/macro.debug_assert.html) if you want that functionality). Here it is just in a test, so not part the main application though, so doesn't really matter either way.

**Comment by ethanpailes** [comment]
> On more comment, should we add a line to test we're getting zero exit status somewhere

That's what the change to the `default_keybinding_detach` test does (when it checks we get a success error code).

File: shpool/tests/attach.rs (hunk 1 above)
Comment (jrmlhermitte) [reply shpool/tests/attach.rs:558]: ah yes that makes sense, sg!

File: shpool/tests/detach.rs (hunk 2 above)
Comment (jrmlhermitte) [reply shpool/tests/detach.rs:38]: im seeing the return type as `Option<int32>` though, is that correct?

here: https://github.com/shell-pool/shpool/blob/55be263b397ad3cebed9e7a7d0d413603bc16a6f/libshpool/src/daemon/exit_notify.rs#L40
